src/communication/stream_transfer_sender.hpp

```diff
@@ -36,6 +36,13 @@ namespace cubthread
 namespace cubstream
 {
 
+  /* Interface used by stream senders/receivers to notify threads. */
+  class stream_ack
```

Comment:
Creating a class just for one interface method seems a bit stretched.
What about adding the "notify_stream_ack" method in the base class 'group_complete_manager'.

Reply:
I don't think is good to add notify_stream_ack in group_complete_manager. notify_stream_ack has meaning only in HA context, while group_complete_manager contains common behavior (HA, single node CS, SA)

Reply:
Let's postpone this. The design regarding notify_stream_ack may be discussed and stabilized after control communication channel and ack receivers are implemented.